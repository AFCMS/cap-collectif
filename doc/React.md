### How to write localizable components

Just import the appropriate [component](https://github.com/yahoo/react-intl/wiki#the-react-intl-module) from `react-intl`

- For localizable text use
[`<FormattedMessage>`](https://github.com/yahoo/react-intl/wiki/Components#formattedmessage).

- For date and time:
[`<FormattedDate>`](https://github.com/yahoo/react-intl/wiki/Components#formatteddate)
[`<FormattedTime>`](https://github.com/yahoo/react-intl/wiki/Components#formattedtime)
[`<FormattedRelative>`](https://github.com/yahoo/react-intl/wiki/Components#formattedrelative)

- For numbers and currencies:
[`<FormattedNumber>`](https://github.com/yahoo/react-intl/wiki/Components#formattednumber)
[`<FormattedPlural>`](https://github.com/yahoo/react-intl/wiki/Components#formattedplural)

- If possible, do not use `<FormattedHTMLMessage>`, see how to use *Rich Text Formatting* with
[`<FormattedMessage>`](https://github.com/yahoo/react-intl/wiki/Components#formattedmessage)

- When you need an imperative formatting API, use the [`injectIntl`](https://github.com/yahoo/react-intl/wiki/API#injectintl) High-Order Component.
