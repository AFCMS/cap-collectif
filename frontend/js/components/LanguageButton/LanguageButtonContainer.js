// @flow
import React from 'react';
import { graphql, QueryRenderer } from 'react-relay';
import environment, { graphqlError } from '~/createRelayEnvironment';

import type { LanguageButtonContainerQueryResponse } from '~relay/LanguageButtonContainerQuery.graphql';

import LanguageButtonWrapper from './LanguageButtonWrapper';

const component = ({
  error,
  props,
}: {
  ...ReactRelayReadyState,
  props: ?LanguageButtonContainerQueryResponse,
}) => {
  if (error) {
    return graphqlError;
  }
  if (props) {
    return props.availableLocales && <LanguageButtonWrapper languages={props.availableLocales} />;
  }

  return null;
};

export const LanguageButtonContainer = () => {
  return (
    <QueryRenderer
      environment={environment}
      query={graphql`
        query LanguageButtonContainerQuery {
          availableLocales(includeDisabled: false) {
            ...LanguageButtonWrapper_languages
          }
        }
      `}
      variables={{}}
      render={component}
    />
  );
};

export default LanguageButtonContainer;
