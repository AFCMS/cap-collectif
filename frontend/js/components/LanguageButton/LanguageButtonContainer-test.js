// @flow
/* eslint-env jest */
import React from 'react';
import { shallow } from 'enzyme';

import { LanguageButtonContainer } from './LanguageButtonContainer';

describe('<LanguageButtonContainer />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<LanguageButtonContainer />);
    expect(wrapper).toMatchSnapshot();
  });
});
