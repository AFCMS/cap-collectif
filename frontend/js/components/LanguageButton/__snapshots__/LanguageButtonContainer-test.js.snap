// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LanguageButtonContainer /> should render correctly 1`] = `
<ReactRelayQueryRenderer
  environment={"RelayModernEnvironment()"}
  query={[Function]}
  render={[Function]}
  variables={Object {}}
/>
`;
